### PR TITLE
Feature/pokemon detail

### PR DESCRIPTION
**Pull request by joanafbrito**
# Description drawing

- Add the code that Hayley did
- Use Hooks to access the api on the component
- Add Hayley Css file
- Successfully finding the PokemonDetails but only at the first search- looks like it is not going to the useEffect more than once 🤔- not sure why- I have commented my code where I need help
( we can change that for a class component  if we think that would be better )

- to be able to render all the info at once I had to add an if condition before the return 🤔

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactor feature (non-breaking change that modifies existing work)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

- [x] local host 3000
- [x] console.log()
- [x] dev tools
- [ ] cypress testing

# Checklist:

- [ ] My code follows the style guidelines of this project/linted
- [ ] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/PokemonDetails/PokemonDetails.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  const [error, setError] = useState("");
+
+  const pokemonImage = getPokemonImage(
+    foundPokemon[0].url.replace(/\D/g, "").slice(1)
+  );
+
+  const getPokemonDetails = async () => {
+
+    let pokemonId = foundPokemon[0].url.replace(/\D/g, "").slice(1);
+    const url = `https://pokeapi.co/api/v2/pokemon/${pokemonId}`;
```
Comment (Matt-Roden) [inline src/components/PokemonDetails/PokemonDetails.js]: I'm confused on why we would try to use the ID to append to the end of the URL when the name works as well. It was my understanding that foundPokemon is an object with a name and a url. So, we can just append the name, or fetch the whole URL from the object. I don't think we need to take the extra time to slice out the ID and append it in when we have the information that is already needed to make this fetch.

Comment (Matt-Roden) [reply src/components/PokemonDetails/PokemonDetails.js]: Example: 
https://pokeapi.co/api/v2/pokemon/${pokemonId} OR
https://pokeapi.co/api/v2/pokemon/${pokemonName}  --> both fetch the exact same data.

File: src/components/PokemonDetails/PokemonDetails.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+      setPokemonDetails(pokemonDetails);
+    } catch (error) {
+      setError(error.message);
+    }
+  };
+// for some reason I don't know yet --it is not reseting the state when we type a diferent name or id 
+// i'm not sure if only clearing the search it is what we need, probably we need a way to look into the details again..
+  useEffect(() => {
+    getPokemonDetails();
+  }, []);
```
Comment (Matt-Roden) [inline src/components/PokemonDetails/PokemonDetails.js:30]: I'm still unclear on all of the nuances of Hooks, if we change this to a class based with component did mount, I can be more helpful. 
But...
If we _first_, clear out the state of foundPokemon in app from this useEffect, and _then_ make the fetch, that could solve this issue.

File: src/components/App/App.js [hunk 3]
```diff
@@ -13,90 +14,96 @@ class App extends Component {
...
       }
     });
 
-    console.log(verifiedName)
+    console.log(verifiedName);
 
     if (verifiedName === undefined) {
-      return this.setState({ error: "Not  a valid name, try again" })
-      console.log('No Good Name!')
+      return this.setState({ error: "Not  a valid name, try again" });
```
Comment (Matt-Roden) [inline src/components/App/App.js:58]: Maybe we change this to a binary (true or false) and we can say, if error is true, render some message or maybe even a searchError component?

**Review by Matt-Roden — COMMENTED** [review]
I think we are getting closer, but really need to make the appropriate changes asap to get back on track and start implementing PWA features and animations

File: src/components/App/App.js (hunk 3 above)
Comment (hayleyw7) [reply src/components/App/App.js:58]: Good idea. I think that's a great place to go from the start we have.

File: src/components/PokemonDetails/PokemonDetails.css [hunk 4]
```diff
@@ -0,0 +1,82 @@
+.pokemon-details-page {
```
Comment (hayleyw7) [inline src/components/PokemonDetails/PokemonDetails.css:1]: Thanks for adding in the css I wrote! Definitely needs refactored, but I think it's a great start!

File: src/components/PokemonDetails/PokemonDetails.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
```
Comment (hayleyw7) [inline src/components/PokemonDetails/PokemonDetails.js:31]: Even with the error, I'm glad we found ways to merge the code, and I am happy about where it's going.

**Review by hayleyw7 — APPROVED** [review]
I think it's ready to go into main, so we can all pull the same code locally to work on the bug.

File: src/components/PokemonDetails/PokemonDetails.js [hunk 6]
```diff
@@ -5,14 +5,15 @@ const PokemonDetails = ({ foundPokemon, getPokemonImage }) => {
...
+  let pokemonId = foundPokemon[0].url.replace(/\D/g, "").slice(1)
+  const pokemonImage = getPokemonImage(pokemonId);
 
+//   let pokemonImage =  pokemonDetails.sprites.other.dream_world.front_default
   const getPokemonDetails = async () => {
 
-    let pokemonId = foundPokemon[0].url.replace(/\D/g, "").slice(1);
-    const url = `https://pokeapi.co/api/v2/pokemon/${pokemonId}`;
+    // let pokemonId = foundPokemon[0].url.replace(/\D/g, "").slice(1);
+    let pokemonName = foundPokemon[0].name
```
Comment (Matt-Roden) [inline src/components/PokemonDetails/PokemonDetails.js:15]: Awesome this is much easier to read for a developer looking at our code base :)

**Review by Matt-Roden — APPROVED** [review]

**Pull request merged**